code/datums/status_effects/xeno_buffs.dm

```diff
@@ -142,10 +141,25 @@
 ///Handles the link breaking due to distance
 /datum/status_effect/xeno_psychic_link/proc/handle_dist(datum/source)
 	SIGNAL_HANDLER
-	if(get_dist(owner, target_mob) > link_range)
-		to_chat(target_mob, span_xenowarning("You are too far away from [owner]."))
-		to_chat(owner, span_xenowarning("[target_mob] is too far away."))
-		owner.remove_status_effect(STATUS_EFFECT_XENO_PSYCHIC_LINK)
+	var/within_range_current = get_dist(owner, target_mob) <= link_range
+	if(within_range_current == within_range)
+		return
+	within_range = within_range_current
+	link_toggle(within_range)
+	to_chat(owner, within_range ? span_xenowarning("[target_mob] is within range again.") : span_xenowarning("[target_mob] is too far away."))
```

Comment:
pretty sure that can become very spammy for no real information given

Reply:
Me and Slotbot opted for keeping it this way for the sake of clarification, for both players. If it's a real concern though, we can see about making it less spammy.